ecs-agent/api/appnet/client.go

```diff
@@ -18,12 +18,22 @@ import (
 	"fmt"
 	"net"
 	"net/http"
+
+	prometheus "github.com/prometheus/client_model/go"
 )
 
 type appnetClientCtxKey int
 
-type client struct {
+// AppnetClient is an interface with customized Appnet client that
+// implements the GetStats and DrainInboundConnections
+type AppNetClient interface {
+	GetStats(adminSocketPath string, statsRequest string) (map[string]*prometheus.MetricFamily, error)
+	DrainInboundConnections(adminSocketPath string, drainRequest string) error
+}
+
+type AppNetAgentClient struct {
 	udsHttpClient http.Client
+	AppNetClient
```

Comment:
Is embedding the `AppNetClient` interface here necessary?

Reply:
It is not absolutely necessary, but it is surely a good way to promote the methods implemented by the interface to the enclosing struct, i.e. AppNetAgentClient. We do follow similar pattern in most other cases as well, for example, wsclient:
https://github.com/aws/amazon-ecs-agent/blob/master/ecs-agent/wsclient/client.go#L180
Let me know if you think we should not include it.